File: cocos/renderer/gfx/gfx-command-allocator.ts
import { GFXDevice } from './gfx-device';

export abstract class GFXCommandAllocator {
    
    constructor(device : GFXDevice) {
        this._device = device;
    }

    public abstract initialize() : boolean;
    public abstract destroy();

    protected _device : GFXDevice;
}

File: native/cocos/renderer/core/gfx/GFXCommandAllocator.cc
#include "CoreStd.h"
#include "GFXCommandAllocator.h"

namespace cc {
namespace gfx {

CommandAllocator::CommandAllocator(Device *device)
: GFXObject(ObjectType::COMMAND_ALLOCATOR), _device(device) {
}

CommandAllocator::~CommandAllocator() {
}

} // namespace gfx
} // namespace cc
